### PR TITLE
feature/IO-585_flavors

### PR DESCRIPTION
**Pull request by egor-slatvinskiy-idf**

### CODE REVIEW

**Review by elizaveta-seleznyova-idf — APPROVED** [review]

File: packages/presentation/lib/ui/movie_page/movie_list_widget.dart [hunk 1]
```diff
@@ -40,6 +41,14 @@ class _MovieListWidgetState extends State<MovieListWidget>
     if (widget.rowData == null || widget.data!.isLoading == true) {
```
Comment (vladimir-hodaryonok-idf) [inline packages/presentation/lib/ui/movie_page/movie_list_widget.dart]: bang

File: packages/data/lib/di/injector_data.dart [hunk 2]
```diff
@@ -72,9 +84,17 @@ void _initModuleRepository() {
...
+  if (environmentConfiguration.prod) {
+    GetIt.instance.registerSingleton<HeaderInterceptorTraktApi>(
+      HeaderInterceptorTraktApi(),
+    );
+  }
+  if (environmentConfiguration.sandbox) {
+    GetIt.instance.registerSingleton<SandboxHeaderInterceptorTraktApi>(
+      SandboxHeaderInterceptorTraktApi(),
+    );
+  }
```
Comment (vladimir-hodaryonok-idf) [inline packages/data/lib/di/injector_data.dart]: is there no chances to make only one interceptor for 2 flavors?

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

File: packages/data/lib/di/injector_data.dart (hunk 2 above)
Comment (egor-slatvinskiy-idf) [reply packages/data/lib/di/injector_data.dart]: yes, of course, but if there are more than 2?

**Review by yuliya-klescheva-idf — APPROVED** [review]

**Review by alexei-kruk-idf — APPROVED** [review]

File: packages/data/lib/di/injector_data.dart [hunk 3]
```diff
@@ -21,12 +22,23 @@ void initInjectorData() {
...
+  if (environmentConfiguration.sandbox) {
+    GetIt.instance.registerSingleton<Dio>(
+      _buildSandboxMovieDioTraktApi(
+        GetIt.instance.get<SandboxHeaderInterceptorTraktApi>(),
+      ),
+      instanceName: _traktApi,
+    );
+  }
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/di/injector_data.dart]: if
else if

File: packages/data/lib/di/injector_data.dart (hunk 2 above)
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/di/injector_data.dart]: -//-

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**